Source/WebCore/editing/MarkupAccumulator.cpp

```diff
@@ -201,34 +202,65 @@ String MarkupAccumulator::serializeNodes(Node& targetNode, SerializedNodes root,
 
 void MarkupAccumulator::serializeNodesWithNamespaces(Node& targetNode, SerializedNodes root, const Namespaces* namespaces, Vector<QualifiedName>* tagNamesToSkip)
 {
-    if (tagNamesToSkip && is<Element>(targetNode)) {
-        for (auto& name : *tagNamesToSkip) {
-            if (downcast<Element>(targetNode).hasTagName(name))
-                return;
-        }
-    }
-
-    Namespaces namespaceHash;
+    WTF::Vector<Namespaces> namespaceStack;
```

Comment:
This namespace stack mirrors the behavior of the recursive algorithm and fixes the xml test failures but I'm not happy with pushing a new `HashMap` on every iteration. Working on a more efficient fix.

Reply:
I kept this stack but I'm now storing unique_ptrs which should be more efficient in the common case of not having a namespace